Sources/Signal.swift

```diff
@@ -237,9 +309,73 @@ public final class Signal<Value, Error: Swift.Error> {
 	}
 }
 
-private struct SignalState<Value, Error: Swift.Error> {
-	var observers: Bag<Signal<Value, Error>.Observer> = Bag()
-	var retainedSignal: Signal<Value, Error>?
+/// The status of a `Signal`.
+///
+/// - note: This three-case, multi-payload enum was verified to be a packed
+///         object reference on i386, x86-64, armv7 and armv8 under the Swift
+///         3.0.1 ABI. Should any changes be made to the enum, please first
+///         verify that the change would not cause the enum to spill over the
+///         size of an object reference on all platforms.
```

Comment:
I make a little snippet of `MemoryLayout`, and run them in the iPhone simulator (i386) and a iOS template app (both 32-bit and 64-bit builds) on iPhone SE.

While AArch32 is not exactly the same as ARMv7, they are backwards compatible.